src/courtroom.cpp

```diff
@@ -1870,24 +1877,34 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
   if (f_char_id >= 0)
     f_charname = ao_app->get_showname(char_list.at(f_char_id).name);
 
-  chatlogpiece *temp =
-      new chatlogpiece(f_charname, f_displayname, m_chatmessage[MESSAGE], false,
-                       m_chatmessage[TEXT_COLOR].toInt());
-  ic_chatlog_history.append(*temp);
-  if (ao_app->get_auto_logging_enabled())
-    ao_app->append_to_file(temp->get_full(), ao_app->log_filename, true);
+  if (m_chatmessage[MESSAGE] == " ") // User-created blankpost
+  {
+    m_chatmessage[MESSAGE] = ""; // Turn it into true blankpost
+  }
+  // General blankpost check
+  chatmessage_is_empty = m_chatmessage[MESSAGE] == "";
+
+  if (!chatmessage_is_empty || ic_chatlog_history.isEmpty() || ic_chatlog_history.last().get_message() != "")
+  {
+    chatlogpiece *temp =
+        new chatlogpiece(f_charname, f_displayname, m_chatmessage[MESSAGE], false,
+                         m_chatmessage[TEXT_COLOR].toInt());
+    ic_chatlog_history.append(*temp);
```

Comment:
Frankly never understood why `new` was used here. You should be able to add to a vector without calling `new`:

```suggestion
    chatlogpiece log_entry(f_charname, f_displayname, m_chatmessage[MESSAGE], false,
                           m_chatmessage[TEXT_COLOR].toInt());
    ic_chatlog_history.append(log_entry);
```

Reply:
Fuck accidentally resolved unresolve pls

Reply:
Can do